### ISSUE TITLE
Log Optuna (hyperparameter tuning) runs

### ISSUE THREAD

**Issue by JackCaster** [Azure/MachineLearningNotebooks#1764]
I set up a pipeline. Within the same `PythonScriptStep` I have a _Pytorch-lightning_ training loop and an _Optuna_ hyperparameter tuning loop. If I do not use Optuna, I can log without any issues the training metrics in the Azure ML portal using the [`MLFlowLogger`](https://pytorch-lightning.readthedocs.io/en/stable/extensions/generated/pytorch_lightning.loggers.MLFlowLogger.html) provided by Pytorch-lightning (e.g., I can see the graphs for train/val loss). But, if I do use Optuna to tune the hyperparameters, I cannot log the training metrics of each tuning loop. After the first hyperparameter tuning step is finished I get:
```
mlflow.exceptions.MlflowException: Cannot start run with ID xxxxx because active run ID does not match environment run ID. Make sure --experiment-name or --experiment-id matches experiment set with set_experiment(), or just use command-line arguments
```
I think there is a run ID collision. Is there a way to log the training metrics of each tuning loop in the Azure ML portal? I am thinking, is it possible to create child runs so to have a different run IDs for each tuning loop?

**Comment by santiagxf** [comment]
This may be related to how the integration with Optuna and MLflow works. [According to their documentation](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.MLflowCallback.html), the name of the experiment needs to match the name of the study. Have you checked that? The name of the experiment in a PythonScriptStep is determined when you submit the run/job. On top of that, you may also want to consider configuring Optuna with `nest_trials=True` cause a step in Azure ML will run on its own run. So you want each iteration of the study to get its own run.

We don't have an example with Optuna but we did publish a HyperParameter tunning example using MLflow with HyperOpt library. Most of them work in a similar way: [Hyper-parameter optimization using HyperOpt and nested runs in MLflow](https://github.com/Azure/azureml-examples/blob/main/notebooks/using-mlflow/train-with-mlflow/xgboost_nested_runs.ipynb)

**Comment by JackCaster** [comment]
Thanks! That guide was helpful. I managed to hack together something that works. Briefly, in the main script I declare:
```python
def _objective(trial, experiment_name, tracking_uri, mlflow_run):
   ...

mlflow.set_experiment(experiment_name)
mlflow.set_tracking_uri(tracking_uri)

with mlflow.start_run() as mlflow_run:
   # set up and initiate Optuna runs
  objective = partial(
            _objective,
            experiment_name=experiment_name,
            tracking_uri=tracking_uri,
            mlflow_run=mlflow_run,
        )
   study.optimize(objective, n_trials=...)
```

Then, in the objective function I have:
```python
# Set up Pytorch-lightning logger
logger = MLFlowLogger(
        experiment_name=experiment_name,
        tracking_uri=tracking_uri,
        tags={"mlflow.parentRunId": mlflow_run.info.run_id}
    )
```
where this `tags={"mlflow.parentRunId": mlflow_run.info.run_id}` is critical for nesting the runs.
